store/store_download.go

```diff
@@ -253,6 +259,109 @@ func downloadReqOpts(storeURL *url.URL, cdnHeader string, opts *DownloadOptions)
 	return &reqOptions
 }
 
+type downloadTimeoutError struct {
```

Comment:
We have TransferSpeedMonitoringWriter which avoids using 'download' in the name. We also don't know whether it is a timeout. Maybe this should be named TransferSpeedBelowMinimalError then?

Reply:
yes, or TransferSpeedThresholdError or just TransferSpeedError (with the assumption that high speeds are not a problem)

Reply:
Done, but kept transferSpeedError unexported for now (I would change it if we decide to move it to osutil or elsewhere).